PurchasesTests/Purchasing/OfferingsManagerTests.swift

```diff
@@ -100,24 +100,114 @@ extension OfferingsManagerTests {
         expect(obtainedOfferings).to(beNil())
     }
 
-    func testOfferingsForAppUserIDReturnsUnexpectedBackendErrorIfBadBackendRequest() {
+    func testOfferingsForAppUserIDReturnsConfigurationErrorIfBackendReturnsEmpty() throws {
+        // given
+        mockBackend.stubbedGetOfferingsCompletionResult = ([:], nil)
+        mockOfferingsFactory.emptyOfferings = true
+
+        // when
+        var obtainedOfferings: Offerings?
+        var completionCalled = false
+        var obtainedError: Error?
+        offeringsManager.offerings(appUserID: MockData.anyAppUserID) { maybeOfferings, maybeError in
+            obtainedOfferings = maybeOfferings
+            completionCalled = true
+            obtainedError = maybeError
+        }
+
+        // then
+        expect(completionCalled).toEventually(beTrue())
+        expect(obtainedOfferings).to(beNil())
+        let error = try XCTUnwrap(obtainedError)
+        expect((error as NSError).code) == ErrorCode.configurationError.rawValue
+    }
+
+    func testOfferingsForAppUserIDReturnsConfigurationErrorIfProductsRequestsReturnsEmpty() throws {
+        // given
+        mockBackend.stubbedGetOfferingsCompletionResult = (MockData.anyBackendOfferingsData, nil)
+        mockProductsManager.stubbedProductsCompletionResult = Set()
+
+        // when
+        var obtainedOfferings: Offerings?
+        var completionCalled = false
+        var obtainedError: Error?
+        offeringsManager.offerings(appUserID: MockData.anyAppUserID) { maybeOfferings, maybeError in
+            obtainedOfferings = maybeOfferings
+            completionCalled = true
+            obtainedError = maybeError
+        }
+
+        // then
+        expect(completionCalled).toEventually(beTrue())
+        expect(obtainedOfferings).to(beNil())
+        let error = try XCTUnwrap(obtainedError)
+        expect((error as NSError).code) == ErrorCode.configurationError.rawValue
+    }
+
+
+    func testOfferingsForAppUserIDReturnsUnexpectedBackendResponseIfOfferingsFactoryCantCreateOfferings() throws {
```

Comment:
i.e.: unexpected backend response if we have products + offerings data, but `OfferingsManager` returns `nil` (this would only happen if there's an error parsing the backend response, like if its format is unexpected)